lib/raw_net_capture/raw_http_capture.rb

```diff
@@ -0,0 +1,81 @@
+# encoding: utf-8
+
+class RawHTTPCapture < StringIO
+  attr_reader :transactions
+
+  def initialize
+    super
+    reset
+  end
+
+  def received(data)
+    @raw_received << data
+  end
+
+  def sent(data)
+    # when there are multiple requests on the same connection, E.g. redirection, we want the last one.
+    if @raw_received.length > 0
+      reset
+    end
+
+    @raw_sent << data
+  end
+
+  def transactions
+    @transactions ||= []
+  end
+
+  private
+
+  attr_reader :raw_sent, :raw_received
+
+  def reset
+    @raw_received = StringIO.new
+    @raw_sent = StringIO.new
+    transactions << Transaction.new(self)
+  end
+
+  class Transaction
+    HEADER_BODY_SEPARATOR = "\r\n\r\n"
+    attr_reader :response, :request
+
+    def initialize(capture)
+      @request = Request.new(capture.send(:raw_sent))
```

Comment:
Why not just create the `StringIO` objects in here instead of peeking at the capture, then maybe you don't even have to pass the capture into Transaction